pkg/tsdb/elasticsearch/time_series_query.go

```diff
@@ -396,6 +398,7 @@ func (p *timeSeriesQueryParser) parse(tsdbQuery plugins.DataQuery) ([]*Query, er
 		}
 		alias := model.Get("alias").MustString("")
 		interval := model.Get("interval").MustString("")
+		intervalMs := model.Get("interval").MustInt64(0)
```

Comment:
Don't think this is correct. The value here is string representation of the interval, e.g. `"10s"`. Need to apply something like this 
https://github.com/grafana/grafana/blob/2026a1dfb6a013f255d465fbac2aed1f83564268/pkg/tsdb/interval/interval.go#L92-L103

Reply:
Hm, I think I just wrongly c/p-ed

`intervalMs := model.Get("interval").MustInt64(0)`, needed to be: `intervalMs := model.Get("intervalMs").MustInt64(0)`

and also, I don't know if `intervalMs` is string, since I can see that:
```
if interval == "" {
		intervalMS := queryModel.Get("intervalMs").MustInt(0)
		if intervalMS != 0 {
			return time.Duration(intervalMS) * time.Millisecond, nil
		}
	}
```

Reply:
In addition, the code you mentioned above, is still used in `calculator.go`

Reply:
I was referring to if you skip introducing `intervalMs := model.Get("interval").MustInt64(0)` and use the existing interval string representation (`interval := model.Get("interval").MustString("")`). 

Then change the `tsdb.GetIntervalFrom` to 
```go
func GetIntervalFrom(interval, timeInterval string, queryInterval string, defaultInterval time.Duration) (time.Duration, error)
```

And `GetMinInterval` to 
```go
GetMinInterval(queryInterval string) (time.Duration, error)
```

> In addition, the code you mentioned above, is still used in calculator.go

Might have missed that part, great. But seems like both interval string and intervalMs integer is pass around now?

I might be confused/misunderstood something :smile: Let me know if my suggestions make sense?

Reply:
`intervalMs` is actually `int64`, since it's milliseconds, and it's different than `interval` which is `string`. So if the plan is to move away from `simpleJson` we need to declare it explicitly I think. Let's discuss more offline.

Reply:
Sorry I'm super confused now. Yes, lets discuss it.

> So if the plan is to move away from simpleJson we need to declare it explicitly I think

My whole point is to explicitly use a string rather than an int64 (as you've seem to advise).

For queries originating from old alerting system the property "interval" of string is part of the query model. To remove usage of simplejson here you should probable decode the JSON into `Query` using `json.Unmarshal`.